File: rookieone/ruby/code/day1/part1.rb
class Day1Part1
  def self.solve(numbers)
    total_increases = 0
    previous_depth = nil

    numbers.each do |depth|
      if previous_depth != nil && previous_depth < depth
        total_increases += 1
      end

      previous_depth = depth
    end
    
    total_increases
  end
end

File: rookieone/elixir/advent_of_code_2021/lib/day1/part1.ex
defmodule AdventOfCode2021.Day1.Part1 do
  @moduledoc """
  Documentation Day 1 Part 1 of Advent of Code
  """

  def measure_depth(input) do
    check_previous_depth(input, nil, 0)
  end

  def check_previous_depth([], _previous_depth, total_increases) do
    total_increases
  end

  def check_previous_depth([depth|depths], nil, total_increases) do
    check_previous_depth(depths, depth, total_increases)
  end

  def check_previous_depth([depth|depths], previous_depth, total_increases) when depth > previous_depth do
    check_previous_depth(depths, depth, total_increases + 1)
  end

  def check_previous_depth([depth|depths], previous_depth, total_increases) do
    check_previous_depth(depths, depth, total_increases)
  end
end
